prime-dotnet-webapi/Controllers/SitesController.cs

```diff
@@ -1161,26 +1202,37 @@ public async Task<ActionResult> FlagSite(int siteId, FromBodyData<bool> flagged)
             return Ok(site);
         }
 
-        // GET: api/sites/pec-exists
+        // GET: api/sites/1/pec-exists
         /// <summary>
-        /// Check if a given PEC already exists, only applicable to non health authority site
+        /// Check if a given PEC exists within non health authority sites, and
+        /// excludes the PEC if already assigned to the site.
         /// </summary>
+        /// <param name="siteId"></param>
         /// <param name="pec"></param>
         /// <returns></returns>
-        [HttpGet("pec-exists", Name = nameof(PecExists))]
+        [HttpGet("{siteId}/pec-exists", Name = nameof(PecExists))]
         [ProducesResponseType(StatusCodes.Status401Unauthorized)]
         [ProducesResponseType(StatusCodes.Status403Forbidden)]
         [ProducesResponseType(typeof(ApiMessageResponse), StatusCodes.Status200OK)]
         [ProducesResponseType(typeof(ApiMessageResponse), StatusCodes.Status400BadRequest)]
-        public async Task<ActionResult> PecExists(string pec)
+        public async Task<ActionResult> PecExists(int siteId, string pec)
```

Comment:
Going to change the name of this method.  This is async validator to check uniqueness.